src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -143,26 +183,7 @@ protected override void StartWorking()
 
         private void OnSourceProviderSourcesChanged(object sender, EventArgs e)
         {
-            if (!this.IsForeground())
-            {
-                this.InvokeBelowInputPriorityAsync(() => OnSourceProviderSourcesChanged(sender, e));
-                return;
-            }
-
-            this.AssertIsForeground();
-
-            try
-            {
-                PackageSources = _packageServices.GetSources(includeUnOfficial: true, includeDisabled: false)
-                    .Select(r => new PackageSource(r.Key, r.Value))
-                    .ToImmutableArrayOrEmpty();
-            }
-            catch (Exception ex) when (ex is InvalidDataException || ex is InvalidOperationException)
-            {
-                // These exceptions can happen when the nuget.config file is broken.
-                PackageSources = ImmutableArray<PackageSource>.Empty;
-            }
-
+            _packageSources = default;
```

Comment:
whole purpose of this service is so that we can get sources from this service and save it so that code fix doesn't need to call nuget directly. 

I think if we are going to embrace JTF (I do want us to if others are okay), we need to change code fix to use JTF.Run rather than Task.Wait or Result.

Reply:
> whole purpose of this service is so that we can get sources from this service and save it so that code fix doesn't need to call nuget directly.

This is fine if it occurs OOP. Currently it does not, so we opt to instead delay the load hit to the first time the feature is used.

Reply:
Resolving this item as a duplicate of https://github.com/dotnet/roslyn/pull/35364#discussion_r279846787

Reply:
ya. since the API can call from BG, there is no issue. but since APIs can call from BG, sounds like everything can move to BG though. nothing require UI thread then.